docs/sql-data-sources-csv.md

```diff
@@ -109,9 +109,9 @@ Data source options of CSV can be set via:
     <td>read</td>
   </tr>
   <tr>
-    <td><code>inferDate</code></td> 
+    <td><code>prefersDate</code></td>
     <td>false</td>
-    <td>Whether or not to infer columns that satisfy the <code>dateFormat</code> option as <code>Date</code>. Requires <code>inferSchema</code> to be <code>true</code>. When <code>false</code>, columns with dates will be inferred as <code>String</code> (or as <code>Timestamp</code> if it fits the <code>timestampFormat</code>).</td>
+    <td>Attempts to infer string columns as <code>Date</code> if the values satisfy <code>dateFormat</code> option and failed to be parsed by the respective formatter during schema inference (<code>inferSchema</code>). When used in conjunction with a user-provided schema, attempts to parse timestamp columns as dates using <code>dateFormat</code> if they fail to conform to <code>timestampFormat</code>, the parsed values will be cast to timestamp type afterwards.</td>
```

Comment:
```suggestion
    <td>During schema inference (<code>inferSchema</code>), attempts to infer string columns that contain dates or timestamps as <code>Date</code> if the values satisfy the <code>dateFormat</code> option and failed to be parsed by the respective formatter. Attempts to parse timestamp columns as dates using <code>dateFormat</code> if they fail to conform to <code>timestampFormat</code>. The parsed values will be cast to timestamp type afterwards.</td>
```

1. Re-order sentence for readability
2. Removed: "When used in conjunction with a user-provided schema" because parsing timestamp columns as dates happens regardless of whether there is a user-provided schema
3. Split sentence at comma for readabilityWhen used in conjunction with a user-provided schema

Reply:
I would still prefer to keep the change for user-provided schema. There is a different behaviour when schema inference or providing one. During the former, string columns would be converted to dates but in the latter they would be kept as timestamp columns.

Reply:
Let me update and we can discuss further.

Reply:
I updated the text, essentially kept your suggestions and just inserted "With a user-provided schema". Can you take another look? Thanks.

Reply:
I see, that makes sense. In my opinion, it is more confusing for the reader. But LGTM

Reply:
Yes, thanks. I just wanted to clarify the behaviour around user-provided schema as opposed to schema inference to remove any ambiguity that would arise from users.